### ISSUE TITLE
Migration issue with .NET 7

### ISSUE THREAD

**Issue by ssougnez** [danielgerlag/workflow-core#1128]
**Describe the bug**

I currently have an application running with .NET 5 and wfc 3.4.0. I've been working on updating its dependencies and everything went fine until .NET 7. Indeed, when upgrading to .NET 6 and wfc 3.8.1, everything went as expected. However, today, I tried the same upgrade but towards .NET 7 but the migration of wfc failed with the following error:

> System.AggregateException: 'One or more errors occurred. (Method not found: 'Microsoft.EntityFrameworkCore.Migrations.Operations.Builders.OperationBuilder`1<Microsoft.EntityFrameworkCore.Migrations.Operations.CreateIndexOperation> Microsoft.EntityFrameworkCore.Migrations.MigrationBuilder.CreateIndex(System.String, System.String, System.String[], System.String, Boolean, System.String)'.)'

To pint point the issue, I first tried to upgrade from ".NET 5/wfc 3.4.0" to ".NET 6/wfc 3.8.1" then to ".NET 7/wfc 3.8.1" and with this approach it worked. I guess that the failing migration could be applied with .NET 6 but not with .NET 7.

I'm OK to deploy the application in two steps to fix this issue but I'm wondering whether this issue will occur again in the future. What if you create your database migrations with an old version of EF Core generating code that is not compatible starting with .NET 7? I'm just shooting in the dark here, but I think this issue should be investigated to ensure that future updates won't break.

**To Reproduce**

- Use wfc 3.4.0 with .NET 5
- Upgrade to wfc 3.8.1 with .NET 7

**Expected behavior**

Migrations should be applied without any issue.

**Additional contact**

I also use:

- WorkflowCore.Persistence.SqlServer 3.8.1
- WorkflowCore.Providers.Redis 3.8.1

**Comment by danielgerlag** [comment]
I don't think this is a .NET 7 issue... you should be able to use .NET 7 with EF Core 6....
It is an EF Core issue, the culture of the EF Core team is to make breaking changes from version to version.

**Comment by ssougnez** [comment]
But if this is the case, it might mean that the SQL provider wouldn't work on a new project using .NET 7 because of the past migrations...

It's a weird one... I'll do the upgrade in two times but I feared that a future migration might fail because of this.

**Comment by danielgerlag** [comment]
I regret having used EF core for several of the DB providers... I want to say we should start thinking about rewriting them without EF Core.

**Comment by ssougnez** [comment]
Indeed, but that's not an easy/quick task...

**Comment by YulianaPoliakova** [comment]
I had the same issue with migrating from .NET 6/wfc 3.4.0 to .NET 7/wfc 3.8.1. 

Migrating first the wfc from .NET 6/3.4.0 to .NET 6/3.81 and then .NET 6/3.81 to  .NET 7/3.81 helped.

**Comment by HermanSchoenfeld** [comment]
> I regret having used EF core for several of the DB providers... I want to say we should start thinking about rewriting them without EF Core.

I have an NHibernate mapping of your schema. Only thing missing is implementation of the repositories. If interested please I can send them.

**Comment by IvanJosipovic** [comment]
I just faced this issue, my temporary fix was to downgrade EF to 6 by adding the following package references:

```
		<PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.13" />
		<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.13" />
		<PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.13" />
```

Also check any of your dependencies and make sure they don't have  any other Microsoft.EntityFrameworkCore.* upgraded to 7, if so try adding a manual reference to version 6 of that package.


Overall, I think the following solutions could work:
- Upgrade workflow core to use EF version 7
- create separate versioned EF packages.
  - WorkflowCore.Persistence.SqlServer6
  - WorkflowCore.Persistence.SqlServer7
  - etc

**Comment by ThomasBergholdWieser** [comment]
The fact that this issue goes months without being addressed by the package author as well as the lack of continued documentation is worrying.

**Comment by elexisvenator** [comment]
We have ran into this issue, with an application that is also using [Marten](https://martendb.io/).  The catch here is that marten 6 requires npgsql7, and the efcore interface for npgsql7 also requires efcore 7. So we are stuck.

I am happy to push through an upgrade for workflowcore, though it really depends on which approach the team wants to support (previous post)?  Is the likely approach going forward to upgrade workflowcore persistence to efcore 7, or to create a efcore 7 variant package?

**Comment by ssougnez** [comment]
Any update for this issue?

For information, I tried using this package with a new .NET 7 application along with the EF Core provider and it does not even start (because of the same issue as in the OP).

This means that WorkflowCore with EF Core provider is not compatible with .NET 7.
